song_rater/tests.py

```diff
@@ -29,3 +29,16 @@ def test_add_song_post(self):
     def test_add_song_incomplete_post(self):
         response = self.client.post('/add-song', {'artist': 'Orange'})
         self.assertEqual(response.status_code, 200)
+
+
+class SongListTestCase(TestCase):
+    def setUp(self):
+        song = Song.objects.create(artist='Apple', title='Pie')
+        song.ratings.create(rating=3)
+
+    def test_song_list(self):
+        response = self.client.get('/song-list')
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, 'Apple', count=1)
+        self.assertContains(response, 'Pie', count=1)
+        self.assertContains(response, '3', count=1)
```

Comment:
Would it be interesting to test another boundary condition: How your song listing works if there are *no* elements in the song table yet?

It may also be a good idea to test a listing with multiple songs: Does that work? (I see no reason for why it wouldn't work, but it tests the situation in which returning one song works, but the listing fails with multiple items.)

Reply:
I added tests for both of these cases. Thank you